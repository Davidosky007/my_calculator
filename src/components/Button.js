/* eslint-disable linebreak-style */
import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ name, handleClick, className }) => (
  <div>
    <button type="button" onClick={() => handleClick(name)} className={className}>
      { name }
      {' '}
    </button>
  </div>
);

Button.propTypes = {
  name: PropTypes.string,
  handleClick: PropTypes.func,
  className: PropTypes.string,
};

Button.defaultProps = {
  name: null,
  handleClick: null,
  className: 'btn button',
};

export default Button;
