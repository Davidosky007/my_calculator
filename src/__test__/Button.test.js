/* eslint-disable linebreak-style */
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import '@testing-library/jest-dom';
import Button from '../components/Button';

describe('Button component test', () => {
  test('render Button component correctly', () => {
    const button = create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  });
});
