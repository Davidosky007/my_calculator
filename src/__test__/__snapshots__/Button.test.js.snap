// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test render Button component correctly 1`] = `
<button
  className="btn button"
  onClick={[Function]}
  type="button"
>
   
</button>
`;
